libs/Microsoft.MixedReality.WebRTC.Unity/Assets/Microsoft.MixedReality.WebRTC.Unity/Scripts/PeerConnection.cs

```diff
@@ -645,8 +645,27 @@ public MediaLine AddTransceiver(MediaKind kind)
 
         /// <summary>
         /// Create a new connection offer, either for a first connection to the remote peer, or for
-        /// renegotiating some new or removed tracks.
+        /// renegotiating some new or removed transceivers.
+        /// 
+        /// This method submits an internal task to create an SDP offer message. Once the message is
+        /// created, the implementation raises the <see xref="Microsoft.MixedReality.WebRTC.PeerConnection.LocalSdpReadytoSend"/>
+        /// event to allow the user to send the message via the chosen signaling solution to the remote
+        /// peer.
+        /// 
+        /// <div class="IMPORTANT alert alert-important">
+        /// <h5>IMPORTANT</h5>
+        /// <p>
+        /// This method is very similar to the method available in the underlying C# library, and
+        /// actually calls it. However it also performs additional work in order to pair the transceivers
+        /// of the local and remote peer. Therefore Unity applications must call this overload instead
+        /// of the base C# library one to ensure transceiver pairing works as intended.
```

Comment:
nit: all these disclaimers make me think that it might be worth considering calling these methods differently.

Reply:
That will not change things much IMHO. Users looking at the Unity scripts will do the right things, while users digging into the C# library API are at risk of breaking the Unity integration. Ideally we'd hide the C# peer, as already discussed, but that prevents access to features not explicitly exposed in Unity.

Reply:
Yes, I agree there is no way to make everything 100% safe, just saying that by using the exact same names we might be making things worse than necessary without a strong reason.

Reply:
What name are you suggesting? How about:
- `StartConnectionAsync()` == `CreateOffer[Async]()`
- `ApplyRemoteOfferAsync()`
- `ApplyRemoteAnswerAsync()`

...with the last 2 ones calling `SetRemoteDescriptionAsync()` from the C# library?

Reply:
Or even `ConnectAsync()` to make it even more simple?

Reply:
"Connect" sounds good. Splitting offer and answer handling might bring to duplicating some code paths so I'd rather leave them one method. Maybe `HandleConnectionMessage`?

Reply:
Fixed. As discussed offline, kept `StartConnection()` until we refactor `CreateOffer()` to be a proper C# async method.